src/Features/Core/Portable/CodeRefactorings/CodeRefactoringService.cs

```diff
@@ -183,11 +186,16 @@ private ConcatImmutableArray<CodeRefactoringProvider> GetProviders(Document docu
                 var task = provider.ComputeRefactoringsAsync(context) ?? Task.CompletedTask;
                 await task.ConfigureAwait(false);
 
-                var result = actions.Count > 0
-                    ? new CodeRefactoring(provider, actions.ToImmutable())
-                    : null;
-
-                return result;
+                if (actions.Count > 0)
+                {
+                    var fixAllProviderInfo = extensionManager.PerformFunction(
+                        provider, () => ImmutableInterlocked.GetOrAdd(ref _fixAllProviderMap, provider, FixAllProviderInfo.Create), defaultValue: null);
+                    return new CodeRefactoring(provider, actions.ToImmutable(), fixAllProviderInfo);
+                }
+                else
+                {
+                    return null;
+                }
```

Comment:
nit: for the "early-return" pattern, i prefer inverting the if, so that hte bailout cases are up front, and easy to relate to the boolean condition.  tehn, the complex code comes afterwards.